### ISSUE TITLE
Watchdog not working at paths containing spaces

### ISSUE THREAD

**Issue by complanar** [Cimbali/pympress#203]
### Environment
- Linux Mint 20.1
- Python 3.8.5

### Pympress
- Pympress version: 1.6.1
- Installation method: source

### Expected behaviour
Pympress does update the view of an updated PDF file.

### Steps to reproduce
Open a PDF file in pympress (at path containing spaces, see below), update the file in the background

### Problem
Pympress does not update its view of the updated file

What is reported in pympress.log?
```
INFO:pympress.app:Pympress: 1.6.1 ; Python: 3.8.5 ; OS: Linux 5.4.0-77-generic #86-Ubuntu SMP Thu Jun 17 02:35:03 UTC 2021 ; Gtk 3.24.20 ; GLib 2.64.2 ; Poppler 0.86.1 cairo ; Cairo 1.16.0 , pycairo 1.16.2
INFO:pympress.extras:Media support: GdkPixbuf gif player, GStreamer 1.16.2, VLC 3.0.9.2 Vetinari
WARNING:pympress.ui:Gtk.Application.inhibit failed preventing screensaver, trying hard disabling
WARNING:pympress.util:Should not require hard enable/disable screensaver on Linux
ERROR:pympress.extras:Impossible to open dir at /home/teichert/Schule/Informatik/Material/4%20Rekursion%20und%20Iteration
Traceback (most recent call last):
  File "/home/teichert/Development/pympress/pympress/extras.py", line 814, in watch_file
    self.observer.schedule(self.monitor, directory)
  File "/usr/lib/python3/dist-packages/watchdog/observers/api.py", line 293, in schedule
    emitter.start()
  File "/usr/lib/python3/dist-packages/watchdog/utils/__init__.py", line 110, in start
    self.on_thread_start()
  File "/usr/lib/python3/dist-packages/watchdog/observers/inotify.py", line 121, in on_thread_start
    self._inotify = InotifyBuffer(path, self.watch.is_recursive)
  File "/usr/lib/python3/dist-packages/watchdog/observers/inotify_buffer.py", line 35, in __init__
    self._inotify = Inotify(path, recursive)
  File "/usr/lib/python3/dist-packages/watchdog/observers/inotify_c.py", line 199, in __init__
    self._add_dir_watch(path, recursive, event_mask)
  File "/usr/lib/python3/dist-packages/watchdog/observers/inotify_c.py", line 379, in _add_dir_watch
    raise OSError('Path is not a directory')
OSError: Path is not a directory
```

Does the problem still happen if you remove your config file?
Yes, it does.

Maybe the problem is about path names containing spaces, but the watchdog mechanism does not reliably work either, if the path does not contain any spaces. On one of my devices it does work, on the other with identical setup it does not. However, when opening a file named without spaces the log file is always empty, except of the first four lines.

**Comment by Cimbali** [comment]
Does it change based on how you open the file? i.e. from the command line or open dialog?

**Comment by complanar** [comment]
Yes, I did not try both ways. I am normally opening the file from the command line. When I am doing this, the error message in the log file appears already at startup.

When I open the same file with the open dialog, everything works as expected. No error message and reloading as intended. This might be a workaround in the meantime. Thanks for pointing me in the right direction.

**Comment by Cimbali** [comment]
I still need to test this on Windows as I’m not sure how this all interacts with the different path separator.

**Comment by complanar** [comment]
Thank you for your work. 
Unfortunately I am not sure, if your changes do work the way they should. Maybe its me, who is doing something wrong.

I just fetched your changes to my local pympress repository reinstalled it from source but now the second screen blanks when I change a file while its open and the presenter view looks as if no file was opened except that the clock is still ticking.

The log file is completely empty. No difference if I open the file with the open dialog or by the command line. Deleted the config file - no changes. Opened different files - the same. Tested if it is LaTeX. But if I modify a pdf file with LibreOffice the same happens.

Is there any debug switch for the command line that could help?

**Comment by complanar** [comment]
Sorry just found the debug level switch. Will investigate further.

**Comment by complanar** [comment]
Does this help?

```
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'/home/teichert/Schule/.~lock.Eingangspr\xc3\xa4sentation Videokonferenz.pdf#', wd=1, mask=IN_CREATE, cookie=0, name=b'.~lock.Eingangspr\xc3\xa4sentation Videokonferenz.pdf#'>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'/home/teichert/Schule/.~lock.Eingangspr\xc3\xa4sentation Videokonferenz.pdf#', wd=1, mask=IN_MODIFY, cookie=0, name=b'.~lock.Eingangspr\xc3\xa4sentation Videokonferenz.pdf#'>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'/home/teichert/Schule/lu31104f99751.tmp', wd=1, mask=IN_CREATE, cookie=0, name=b'lu31104f99751.tmp'>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'/home/teichert/Schule/lu31104f99751.tmp', wd=1, mask=IN_MODIFY, cookie=0, name=b'lu31104f99751.tmp'>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'/home/teichert/Schule/lu31104f99751.tmp', wd=1, mask=IN_MODIFY, cookie=0, name=b'lu31104f99751.tmp'>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'/home/teichert/Schule/lu31104f99751.tmp', wd=1, mask=IN_MOVED_FROM, cookie=4112, name=b'lu31104f99751.tmp'>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'/home/teichert/Schule/Eingangspr\xc3\xa4sentation Videokonferenz.pdf', wd=1, mask=IN_MOVED_TO, cookie=4112, name=b'Eingangspr\xc3\xa4sentation Videokonferenz.pdf'>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'/home/teichert/Schule/Eingangspr\xc3\xa4sentation Videokonferenz.pdf', wd=1, mask=IN_ATTRIB, cookie=0, name=b'Eingangspr\xc3\xa4sentation Videokonferenz.pdf'>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'/home/teichert/Schule/.~lock.Eingangspr\xc3\xa4sentation Videokonferenz.pdf#', wd=1, mask=IN_DELETE, cookie=0, name=b'.~lock.Eingangspr\xc3\xa4sentation Videokonferenz.pdf#'>
```

Log entries when opening a file and then changing it by LibreOffice.

Opened file is `/home/teichert/Schule/Eingangspr\xc3\xa4sentation Videokonferenz.pdf`. `\xc3\xa4` should be `ä` (german umlaut character), but error is the same with files not containing any special characters.

**Comment by Cimbali** [comment]
`\xc3\xa4` is correct, it’s the utf-8 encoding of `ä`

**Comment by Cimbali** [comment]
Current master should be better now @complanar. Still untested on Windows though.

**Comment by complanar** [comment]
Great! Just tested it. Thank you.

**Comment by Cimbali** [comment]
Tested, and released as v1.6.2
